src/preact/vote-buttons.js

```diff
@@ -21,15 +21,8 @@ import {h, Component} from 'preact';
 
 export default class VoteButtons extends Component {
 
-  constructor (props) {
-    super(props);
-
-    this.state = {
-      score: this.props.score
-    };
-
-    this.onVoteUp = this.onVoteUp.bind(this);
-    this.onVoteDown = this.onVoteDown.bind(this);
+  componentWillMount () {
+    this.state.score = this.props.score;
   }
 
```

Comment:
@simevidas not to self-plug, but you might be interested in taking a look at the [`@bind` method decorator from decko](https://github.com/developit/decko#bind). It's similar to autobind-decorator but smaller, per-method, lazily bound and uses closures instead of `.bind()` (much cheaper in anything but current chrome).

I think VoteButtons was stateful in the demonstration in order to create a nontrivial tree of components.  It probably impacts performance, but I'm more interested in solving that within Preact than circumventing it in benchmarks 👍